addon/orm/associations/belongs-to.js

```diff
@@ -72,15 +74,15 @@ class BelongsTo extends Association {
           - sets the associated parent (via model)
       */
       set: function(newModel) {
-        if (newModel && newModel.isNew()) {
-          this[foreignKey] = null;
+        if (newModel && newModel['isNew'] && newModel.isNew()) {
```

Comment:
the `newModel` that's getting sent in (at least sometimes) is a stripped down POJO, so it couldn't call isNew().